src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -5033,24 +5033,24 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         case Select(qual @ Super(_, _), nme.CONSTRUCTOR) =>
           // the qualifier type of a supercall constructor is its first parent class
           typedSelect(tree, typedSelectOrSuperQualifier(qual), nme.CONSTRUCTOR)
+        case Select(qual, name) if name.isTypeName =>
+          val qual1 = typedTypeSelectionQualifier(qual, WildcardType)
+          if (qual1.isErrorTyped) ErrorUtils.issueNormalTypeError(qual1, "Bad path for selection")
```

Comment:
I suspect this error message might be a bit unclear to some users. With hope that I'm not dooming your PR to months of error-message bikeshedding, what about something like:

```
t10695.scala:6: error: Cannot select from method raw: expected a stable path (value or package).
  val node: raw.Node = null
            ^
```

Which still mentions "path", but at least it sheds some light on what the invalid qualifier is (imagining that it is declared elsewhere and its nature is possibly unclear by looking at the vicinity of the error).

Reply:
Yes, it's a pretty hard core message. In the bad cycle t6535, do we know what happened? I think I left it vague enough, intending to rely on the caret. I'd be happy to improve the message. `Obviously erroneous.`

Reply:
+1 on ContextError and a more talking message

Reply:
I wonder if the error should not be reported in `def cantAdapt` (in adapt). cc @adriaanm - related to https://github.com/scala/scala/pull/6334, which is assigned to you for review.

Reply:
I think that could work. We can check `QUALmode` to decide whether this could even be a method application.